src/main/java/com/ft/methodearticleinternalcomponentsmapper/transformation/InternalComponentsMapper.java

```diff
@@ -77,23 +85,32 @@
 
     public InternalComponentsMapper(final FieldTransformer bodyTransformer,
```

Comment:
Could you please create a Map for validators, by source->validator, and merge together the two validators under the same class, they look very similar. Then instead of the if-else, you can use the map to get the correct one.